resources/idlharness.js

```diff
@@ -3288,15 +3292,24 @@ function idl_test(srcs, deps, idl_setup_func) {
         deps = (deps instanceof Array) ? deps : [deps] || [];
         var setup_error = null;
         return Promise.all(
-            srcs.concat(deps).map(function(spec) {
-                return fetch_spec(spec);
-            }))
-            .then(function(idls) {
+            srcs.concat(deps).map(fetch_spec))
+            .then(function(results) {
+                const astArray = results.map(result =>
+                    WebIDL2.parse(result.idl, { sourceName: result.spec })
+                );
+                const validations = WebIDL2.validate(astArray);
```

Comment:
Do we want to validate the dependencies?

Reply:
Yes, as we need to provide the context of the target IDL. The validator won't know which type an argument or a return type references, without providing the dependencies. Ideally the validator could receive the context without validating it, but that use case is not supported as of now.